src/primitives/compression.cpp

```diff
@@ -0,0 +1,358 @@
+#include <primitives/compression.h>
+
+CCompressedTxId::CCompressedTxId(const uint256& txid) : m_block_height(0), m_block_index(0), m_txid(txid) {};
+
+CCompressedTxId::CCompressedTxId(const uint32_t& block_height, const uint32_t& block_index) : m_block_height(block_height), m_block_index(block_index) {};
+
+std::string CCompressedTxId::ToString() const
+{
+    return strprintf("CCompressedTxId(block_height=%u, block_index=%u, txid=%s)",
+    m_block_height,
+    m_block_index,
+    HexStr(m_txid));
+}
+
+CCompressedOutPoint::CCompressedOutPoint(const CCompressedTxId& txid, const uint32_t& n) : m_txid(txid), m_n(n) {}
+
+std::string CCompressedOutPoint::ToString() const
+{
+    return strprintf("CCompressedOutPoint(txid=%s, n=%u)",
+    m_txid.ToString(),
+    m_n);
+}
+
+CCompressedTxIn::CCompressedTxIn(secp256k1_context* ctx, const CTxIn& txin, const CCompressedTxId& txid, const CScript& scriptPubKey) : m_prevout(txid, txin.prevout.n) {
```

Comment:
In 89b52782f8d6aca75d49048da1e137fbf615c0e6:

As mentioned offline, we should avoid passing secp contexts around like this.

Reply:
Strangely enough the secp256k1_context_static worked for my every use case.

Reply:
Oh, nice, perfect. We aren't doing anything with secret data so that's fine.

Reply:
Great!